src/components/controls/PasswordInput/PasswordInput.tsx

```diff
@@ -0,0 +1,109 @@
+'use client';
+
+import React from 'react';
+
+import {Eye, EyeSlash} from '@gravity-ui/icons';
+
+import {useControlledState} from '../../../hooks';
+import {ActionTooltip} from '../../ActionTooltip';
+import {Button} from '../../Button';
+import {ClipboardButton} from '../../ClipboardButton';
+import {Icon} from '../../Icon';
+import {block} from '../../utils/cn';
+import {TextInput} from '../TextInput';
+import type {TextInputProps} from '../TextInput';
+
+import {i18n} from './i18n';
+import {getActionButtonSizeAndIconSize} from './utils';
+
+import './PasswordInput.scss';
+
+const b = block('password-input');
+
+export type PasswordInputProps = Omit<TextInputProps, 'type'> & {
+    /** Show copy button */
+    showCopyButton?: boolean;
+    /** Show reveal button */
+    showRevealButton?: boolean;
+    /** Disable the tooltip for the copy button. The tooltip will not be displayed */
+    hasCopyTooltip?: boolean;
+    /** Disable the tooltip for the reveal button. The tooltip will not be displayed */
+    hasRevealTooltip?: boolean;
+};
+
+export const PasswordInput = (props: PasswordInputProps) => {
+    const {
+        autoComplete,
+        showCopyButton,
+        rightContent,
+        endContent,
+        showRevealButton,
+        size = 'm',
+        hasCopyTooltip = true,
+        hasRevealTooltip = true,
+        controlProps,
+    } = props;
+
+    const [inputValue, setInputValue] = useControlledState(
+        props.value,
+        props.defaultValue ?? '',
+        props.onUpdate,
+    );
+
+    const [hideValue, setHideValue] = React.useState(true);
+
+    const {actionButtonSize, iconSize} = getActionButtonSizeAndIconSize(size);
+
+    const onClick = () => {
+        setHideValue((hideValue) => !hideValue);
+    };
```

Comment:
Let's focus input after click on the buttons or at least prevent default behaviour onMouseDown like [here](https://github.com/gravity-ui/uikit/blob/main/src/components/controls/common/ClearButton/ClearButton.tsx#L52-L54) @amje